dbt/adapters/firebolt/connections.py

```diff
@@ -102,22 +103,21 @@ def make_jdbc_url(self, credentials):
         jdbc_url = os.path.join("jdbc:firebolt://",
                                 credentials.host,
                                 credentials.database)
+        # For both engine name and account, if there's not a value specified
+        # it uses whatever Firebolt has set as default for this DB. So just
+        # fill in url variables that are not None.
+        url_vars = {key:quote(getattr(credentials, key).lower())
+                    for key in ['engine', 'account']
+                    if getattr(credentials, key)
+                   }
+        # If params, then add them, too.
         if credentials.params:
-            jdbc_url += "".join(
-                    map(
-                        lambda kv: "&"
-                        + urllib.parse.quote(kv[0])
-                        + "="
-                        + urllib.parse.quote(kv[1]),
-                        credentials.params.items(),
-                    )
-            )
-        if credentials.engine_name:
-            # If there's not an engine name specified either as an environment
-            # variable or as an engine_name value in profiles.yml, it uses the
-            # engine Firebolt has set as default for this DB.
-            jdbc_url += f"?engine={urllib.parse.quote(credentials.engine_name)}"
-
+            url_vars.update({key:quote(credentials.params.items()[key]).lower()
```

Comment:
Ah, I always forget you could merge dictionaries like that. Kinda hate the way that syntax looks, though. It's weirdly un-Pythonic.

OMG. Are you telling me I used a dictionary comprehension to compose a dictionary from a dictionary? because that's not embarrassing *at all*.

Reply:
Yeah the double splat dict merge does look weird -- if only there was a built-in dict merge thats readable